lib/src/components/NxButton/__tests__/NxButton.test.tsx

```diff
@@ -5,82 +5,66 @@
   });
 
-  it('passes the disabled attribute through', function() {
-    const button = shallow(<NxButton variant="error" disabled>Disabled Button</NxButton>);
-
-    expect(button).toMatchSelector('button.nx-btn.nx-btn--error');
-    expect(button).toBeDisabled();
+  it('sets disabled on the button when the disabled prop is true', function() {
+    expect(quickRender().getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ disabled: undefined }).getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ disabled: false }).getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ disabled: true }).getByRole('button')).toBeDisabled();
   });
 
   it('disabled by class button has aria-disabled true', function() {
-    const button = shallow(<NxButton className="disabled">Disabled Button</NxButton>);
-
-    expect(button).toHaveProp('aria-disabled', true);
+    expect(quickRender().getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ className: 'disabled' }).getByRole('button')).toHaveAttribute('aria-disabled', 'true');
   });
 
-  it('wraps the button in a tooltip if the title is set and the TooltipContext is not set to true', function() {
-    const noTitleNoContext = mount(<NxButton>foo</NxButton>).children(),
-        titleNoContext = mount(<NxButton title="bar">foo</NxButton>).children(),
-        noTitleContext = mount(
-          <TooltipContext.Provider value={true}>
-            <NxButton>foo</NxButton>
-          </TooltipContext.Provider>
-        ).children(),
-        titleContext = mount(
-          <TooltipContext.Provider value={true}>
-            <NxButton title="bar">foo</NxButton>
-          </TooltipContext.Provider>
-        ).children();
-
-    expect(noTitleNoContext).toMatchSelector('button');
-    expect(noTitleNoContext).toHaveProp('title', undefined);
-
-    expect(titleContext).toMatchSelector('button');
-    expect(titleContext).toHaveProp('title', 'bar');
-
-    expect(noTitleContext).toMatchSelector('button');
-    expect(noTitleContext).toHaveProp('title', undefined);
+  it('gives the icon-only button an accessible name when title prop is set', async function() {
+    const iconOnlyBtnNoTitle = quickRender({ variant: 'icon-only' }).getByRole('button'),
+        iconOnlyBtnTitle = quickRender({ variant: 'icon-only', title: 'Alert' }).getByRole('button');
 
-    expect(titleNoContext).toMatchSelector('NxTooltip');
-    expect(titleNoContext).toHaveProp('title', 'bar');
+    await waitFor(() => expect(iconOnlyBtnNoTitle).not.toHaveAccessibleName());
+    await waitFor(() => expect(iconOnlyBtnTitle).toHaveAccessibleName('Alert'));
   });
 
-  it('throws an error when it contains both disabled and title props', function() {
-    expect(() => {
-      shallow(<NxButton variant="icon-only" title="Delete" disabled><NxFontAwesomeIcon icon={faTrashAlt}/></NxButton>);
-    }).toThrow(TypeError);
+  it('throws an error when it contains both disabled and title props', async function() {
```

Comment:
Is it worth testing here that using `className='disabled'` with a title prop doesn't throw an error?

Reply:
Updated here 5926ddaecd1bd00b7e1df5dbcc665c1b5c83268b.